sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,116 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def drift_show(self) -> Tuple[str, Union[str, dict]]:
+        """
+        Show stack drift for a running stack.
+
+        :returns: The stack drift.
+        """
+        try:
+            status = self._get_status()
+        except StackDoesNotExistError:
+            self.logger.info(f"{self.stack.name} - Does not exist.")
+            status = StackStatus.COMPLETE
+            return status
+
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift_status(detection_id)
+
+        if status in ["DETECTION_COMPLETE", "DETECTION_FAILED"]:
+            response = self._describe_stack_resource_drifts()
+            return_value = (self.stack.external_name, response)
+        else:
+            return_value = (self.stack.external_name, status)
+
+        return return_value
+
+    def _wait_for_drift_status(self, detection_id: str) -> str:
+        """
+        Waits for drift detection to complete.
+
+        :param detection_id: The drift detection ID.
+        :returns: The drift status.
+        """
+        timeout = 300
+        elapsed = 0
+
+        while True:
+            if elapsed >= timeout:
+                return "TIMED_OUT"
+
+            self.logger.debug(f"{self.stack.name} - Waiting for drift detection")
+            response = self._describe_stack_drift_detection_status(detection_id)
+
+            status = response["DetectionStatus"]
+            self._log_drift_status(response)
+
+            if status == "DETECTION_IN_PROGRESS":
+                time.sleep(10)
+                elapsed += 10
+            else:
+                return status
+
+    def _log_drift_status(self, response):
+        """
+        Log the drift status while waiting for
+        drift detection to complete.
+        """
+        keys = [
+            "StackDriftDetectionId",
+            "DetectionStatus",
+            "DetectionStatusReason",
+            "StackDriftStatus"
+        ]
+
+        for key in keys:
+            if key in response:
+                self.logger.debug(
+                    f"{self.stack.name} - {key} - {response[key]}"
+                )
```

Comment:
I think it would be better if you logged all four things in a single message rather than four separate log messages.

Reply:
@jfalkenstein Could you propose exactly what you'd like the log entry to look like? That is, currently you would see:

```
[2021-08-06 16:21:25] - vpc - Waiting for drift detection                                                                                                                         [2021-08-06 16:21:25] - vpc - Detecting Stack Drift Detection Status                                                                                                              
[2021-08-06 16:21:25] - vpc - StackDriftDetectionId - 86c17670-f67e-11eb-a5fc-067ab790b156
[2021-08-06 16:21:25] - vpc - DetectionStatus - DETECTION_IN_PROGRESS                    
[2021-08-06 16:21:25] - vpc - DetectionStatusReason - User Initiated
```

What would you expect to see there instead?

Reply:
You know, that is pretty. Never mind.